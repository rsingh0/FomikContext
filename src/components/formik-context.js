import React, { useEffect } from "react";
import { useFormikContext, Formik, Form, Field } from "formik";

const AutoSubmitToken = () => {
  // Grab values and submitForm from context
  const { isSubmitting, resetForm, submitForm, values, setFieldValue } =
    useFormikContext();

  console.log("Values", JSON.stringify(values));
  console.log("isSubmitting", JSON.stringify(isSubmitting));

  function reset(){
    if (values.token.length !== 6) {
      isSubmitting = false;
      resetForm();
      setFieldValue("distance", "25");
    }
  }

  useEffect(() => {
    // Fetch API?

  }, [values]);

  return (
    <div>
      <h1>2-step Verification</h1>
      <p>Please enter the 6 digit code sent to your device</p>
      <Formik
        initialValues={{ token: "" }}
        validate={(values) => {
          const errors = {};
          if (values.token.length < 5) {
            errors.token = "Invalid code. Too short.";
          }
          return errors;
        }}
        onSubmit={(values, actions) => {
          setTimeout(() => {
            alert(JSON.stringify(values, null, 2));
            actions.setSubmitting(false);
          }, 1000);
        }}
      >
        <Form onReset={reset}>
          <Field name={values.token} type="tel" />
        </Form>
      </Formik>
    </div>
  );
};

export { AutoSubmitToken };
