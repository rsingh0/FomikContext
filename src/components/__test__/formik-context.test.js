import React from "react";
import renderer from "react-test-renderer";
import * as Formik from "formik";
import { AutoSubmitToken } from "../formik-context";

describe("Formik Context: Snapshot", () => {
  const useFormikContextMock = jest.spyOn(Formik, "useFormikContext");
  
  beforeEach(async () => {
    useFormikContextMock.mockReturnValue({
      isSubmitting: false,
      resetForm: (nextState) => {
        return null;
      },
      values: { token: "testValue" },
      submitForm: () => new Promise(() => null),
      setFieldValue: (field, value, shouldValidate) => {
        return null;
      },
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("matches snapshot with reset values", () => {
    const tree = renderer.create(<AutoSubmitToken />);
    expect(tree).toMatchSnapshot();
  });
});
