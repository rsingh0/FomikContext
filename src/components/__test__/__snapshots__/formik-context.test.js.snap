// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Formik Context: Snapshot matches snapshot with reset values 1`] = `
<div>
  <h1>
    2-step Verification
  </h1>
  <p>
    Please enter the 6 digit code sent to your device
  </p>
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <input
      name="testValue"
      onBlur={[Function]}
      onChange={[Function]}
      type="tel"
    />
  </form>
</div>
`;
